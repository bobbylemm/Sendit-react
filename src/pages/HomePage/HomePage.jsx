import React from 'react';
import { Link } from 'react-router-dom';
import '../../css/homePage.scss';
import Button from '../../components/styledComponent/Button';
import createOrderPic from '../../images/create-order.jpg';
import editOrderPic from '../../images/maarten-van-den-heuvel-246398-unsplash.jpg';
import trackOrderPic from '../../images/rhys-moult-328651-unsplash.jpg';
import receiveOrderPic from '../../images/receive-parcel.jpg';

const Homepage = () => (
  <div className="hm">
    <main id="slide-container">
      <div id="hero-bg">
        <div className="container">
          <div className="hero-text">
            <h1>Welcome to SendIt</h1>
            <div className="hero-description">
              <p>
                SendIT is a courier service that helps users deliver parcels to
                different destinations.
                <span className="hide-on-small">
                  SendIT provides courier quotes based on weight categories.
                </span>
              </p>
            </div>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/create-parcel">Get started</Link>
            </Button>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/track-parcels">Track deliveries</Link>
            </Button>
          </div>
        </div>
      </div>
      <div id="hero-bg-2">
        <div className="container">
          <div className="hero-text">
            <h1>Welcome to SendIt</h1>
            <div className="hero-description">
              <p>
                SendIT is a courier service that helps users deliver parcels to
                different destinations.{' '}
                <span className="hide-on-small">
                  SendIT provides courier quotes based on weight categories.
                </span>
              </p>
            </div>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/create-parcel">Get started</Link>
            </Button>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/track-parcels">Track deliveries</Link>
            </Button>
          </div>
        </div>
      </div>
      <div id="hero-bg-3">
        <div className="container">
          <div className="hero-text">
            <h1>Welcome to SendIt</h1>
            <div className="hero-description">
              <p>
                SendIT is a courier service that helps users deliver parcels to
                different destinations.{' '}
                <span className="hide-on-small">
                  SendIT provides courier quotes based on weight categories.
                </span>
              </p>
            </div>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/create-parcel">Get started</Link>
            </Button>
            <Button type="button" className="py-3" primary inlineButton>
              <Link to="/track-parcels">Track deliveries</Link>
            </Button>
          </div>
        </div>
      </div>
    </main>
    {/* this is the about us section */}
    <main className="services">
      <div className="container">
        <h3 className="services-title">WHO WE ARE</h3>
        <h2 className="services-header">
          <span className="highlight">Fast and accurate delivery</span> to your
          door steps
        </h2>
        <p>
          Your parcels orders are our number one priority here at SendIt,With
          branches all over the world, SendIt has become a popularly reputable
          courier service that makes its customers/clients their priority
        </p>
      </div>
      <div className="container">
        <div className="how-to-use">
          <div>
            <h2>Create a new delivery</h2>
            <div className="description-cards">
              <img src={createOrderPic} alt="card" />
              <p>
                Generate your parcel delivery, weighing from 100kg - 1000000kg.
              </p>
            </div>
          </div>
          <div>
            <h2>Edit a delivery</h2>
            <div className="description-cards">
              <img src={editOrderPic} alt="edit" />
              <p>
                Edit your parsel delivery, and arrange them according to your
                choice.
              </p>
            </div>
          </div>
        </div>
        {/*  */}
        <div className="how-to-use">
          <div>
            <h2>Track a delivery</h2>
            <div className="description-cards">
              <img src={trackOrderPic} alt="track-card" />
              <p>
                Be aware of your package location at every given point in time
              </p>
            </div>
          </div>
          <div>
            <h2>Recieve delivery on time</h2>
            <div className="description-cards">
              <img src={receiveOrderPic} alt="receive-order-pic" />
              <p>Expect your package at the given time</p>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
);

export default Homepage;
