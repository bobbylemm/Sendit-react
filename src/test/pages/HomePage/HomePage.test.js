import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import HomePage from '../../../pages/HomePage';

describe('Test the snapshot of the home page', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
