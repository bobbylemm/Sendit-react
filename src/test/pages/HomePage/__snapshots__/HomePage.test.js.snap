// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the snapshot of the home page should match the snapshot 1`] = `
<div
  className="hm"
>
  <main
    id="slide-container"
  >
    <div
      id="hero-bg"
    >
      <div
        className="container"
      >
        <div
          className="hero-text"
        >
          <h1>
            Welcome to SendIt
          </h1>
          <div
            className="hero-description"
          >
            <p>
              SendIT is a courier service that helps users deliver parcels to different destinations.
              <span
                className="hide-on-small"
              >
                SendIT provides courier quotes based on weight categories.
              </span>
            </p>
          </div>
          <button
            className="py-3 sc-bdVaJa gasapN"
            type="button"
          >
            <a
              href="/create-parcel"
              onClick={[Function]}
            >
              Get started
            </a>
          </button>
        </div>
      </div>
    </div>
    <div
      id="hero-bg-2"
    >
      <div
        className="container"
      >
        <div
          className="hero-text"
        >
          <h1>
            Welcome to SendIt
          </h1>
          <div
            className="hero-description"
          >
            <p>
              SendIT is a courier service that helps users deliver parcels to different destinations.
               
              <span
                className="hide-on-small"
              >
                SendIT provides courier quotes based on weight categories.
              </span>
            </p>
          </div>
          <button
            className="py-3 sc-bdVaJa gasapN"
            type="button"
          >
            <a
              href="/create-parcel"
              onClick={[Function]}
            >
              Get started
            </a>
          </button>
        </div>
      </div>
    </div>
    <div
      id="hero-bg-3"
    >
      <div
        className="container"
      >
        <div
          className="hero-text"
        >
          <h1>
            Welcome to SendIt
          </h1>
          <div
            className="hero-description"
          >
            <p>
              SendIT is a courier service that helps users deliver parcels to different destinations.
               
              <span
                className="hide-on-small"
              >
                SendIT provides courier quotes based on weight categories.
              </span>
            </p>
          </div>
          <button
            className="py-3 sc-bdVaJa gasapN"
            type="button"
          >
            <a
              href="/create-parcel"
              onClick={[Function]}
            >
              Get started
            </a>
          </button>
        </div>
      </div>
    </div>
  </main>
  <main
    className="services"
  >
    <div
      className="container"
    >
      <h3
        className="services-title"
      >
        WHO WE ARE
      </h3>
      <h2
        className="services-header"
      >
        <span
          className="highlight"
        >
          Fast and accurate delivery
        </span>
         to your door steps
      </h2>
      <p>
        Your parcels orders are our number one priority here at SendIt,With branches all over the world, SendIt has become a popularly reputable courier service that makes its customers/clients their priority
      </p>
    </div>
    <div
      className="container"
    >
      <div
        className="how-to-use"
      >
        <div>
          <h2>
            Create a new delivery
          </h2>
          <div
            className="description-cards"
          >
            <img
              alt="card"
              src="test-file-stub"
            />
            <p>
              Generate your parcel delivery, weighing from 100kg - 1000000kg.
            </p>
          </div>
        </div>
        <div>
          <h2>
            Edit a delivery
          </h2>
          <div
            className="description-cards"
          >
            <img
              alt="edit"
              src="test-file-stub"
            />
            <p>
              Edit your parsel delivery, and arrange them according to your choice.
            </p>
          </div>
        </div>
      </div>
      <div
        className="how-to-use"
      >
        <div>
          <h2>
            Track a delivery
          </h2>
          <div
            className="description-cards"
          >
            <img
              alt="track-card"
              src="test-file-stub"
            />
            <p>
              Be aware of your package location at every given point in time
            </p>
          </div>
        </div>
        <div>
          <h2>
            Recieve delivery on time
          </h2>
          <div
            className="description-cards"
          >
            <img
              alt="receive-order-pic"
              src="test-file-stub"
            />
            <p>
              Expect your package at the given time
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
